changelogs/fragments/4106-proxmox-efidisk0-support.yaml

```
@@ -0,0 +1,5 @@
+---
+minor_changes:
+  - Added efidisk0 support when creating VM with OVMF UEFI BIOS
+bugfixes:
+  - Fixes VM created with OVMF BIOS are invalid (https://github.com/ansible-collections/community.general/issues/1638)
```

```suggestion
  - proxmox_kvm - fixes VM created with OVMF BIOS are invalid (https://github.com/ansible-collections/community.general/issues/1638).
```
A question about this: for this bug to be fixed, does the new option have to be used? Or is it also fixed if the new option is not used? If the former, better add the link to this issue in the `minor_changes` fragment above, and remove this fragment, as this is not a bugfix in the classical sense, but a new feature required to make something work that does not work right now.

---

For the bug to be fixed, although there's room for discussion about it being really a bug (see above comments), the new option must be given. Since the storage for the EFI disk has to be provided by the playbook (i.e. Proxmox VE API can't automatically assign a storage for the EFI disk, from my understanding at least), there cannot really be a default value for this option. Also if I understand correctly the rationale for this module is not to have default options since version 4, so the value has to be provided.